python.mk

```diff
@@ -37,107 +40,132 @@ ${PYTHON} ${ENV_DIR} ${PIP}:
 
+$${$1_WHEELHOUSE}/.clean-wheels: $${$1_REQUIREMENTS} ${ENV_DIR}/.env
+	# Whenever the requirements change, clear out the wheelhouse
+	rm -rf $${$1_WHEELHOUSE}/*
+	mkdir -p $${$1_WHEELHOUSE}
 	touch $$@
 
-# Downloads required dependencies and installs them in the local environment
-${ENV_DIR}/.$1-install-wheels: ${ENV_DIR}/.wheels-built $${$1_REQUIREMENTS}
-	# Download the required dependencies
+$${$1_WHEELHOUSE}/.download-wheels: $${$1_WHEELHOUSE}/.clean-wheels
+  # Download the required dependencies for this component
 	$${$1_FLAGS} ${PIP} wheel -w $${$1_WHEELHOUSE} $$(foreach req,$${$1_REQUIREMENTS},-r $${req}) --find-links $${$1_WHEELHOUSE}
+	touch $$@
 
-	# Install the required dependencies in the local environment
-	${INSTALLER} --find-links=$${$1_WHEELHOUSE} $$(if $${$1_EXTRA_LINKS},--find-links=$${$1_EXTRA_LINKS},) $${$1_WHEELS} $$(foreach req,$${$1_REQUIREMENTS},-r $${req})
-
+# Builds the wheels for this component
+$${$1_WHEELHOUSE}/.build-wheels: $${$1_SETUP} $${$1_SOURCES} $${$1_WHEELHOUSE}/.clean-wheels
+	# For each setup.py file, generate the wheel
+	$$(foreach setup, $${$1_SETUP}, \
+		$${$1_FLAGS} ${PYTHON} $${setup} $$(if $${$1_BUILD_DIRS},build -b ${ROOT}/build_$$(subst .py,,$${setup})) bdist_wheel -d $${$1_WHEELHOUSE} &&) true
 	touch $$@
 
+${ENV_DIR}/.$1-install-wheels: $${$1_WHEELHOUSE}/.download-wheels
+  # Install all wheels in the wheelhouse into the virtual env for this component
+	${INSTALLER} --find-links=$${$1_WHEELHOUSE} $$(if $${$1_EXTRA_LINKS},--find-links=$${$1_EXTRA_LINKS},) $$(foreach req,$${$1_REQUIREMENTS},-r $${req})
+	touch $$@
 
-endef
+${ENV_DIR}/.$1-install-built-wheels: $${$1_WHEELHOUSE}/.download-wheels $${$1_WHEELHOUSE}/.build-wheels
+  # Install all wheels in the wheelhouse into the virtual env for this component
+	${INSTALLER} --find-links=$${$1_WHEELHOUSE} $$(if $${$1_EXTRA_LINKS},--find-links=$${$1_EXTRA_LINKS},) $${$1_WHEELS}
+	touch $$@
 
-# Common test and coverage targets.
-# To use these, the following must be defined:
-#     * TEST_SETUP_PY: list of the setup.py files that are run under coverage
-#
-# The following are optional, but will be used if defined:
-#     * TEST_PYTHON_PATH: the PYTHONPATH used for the tests
-#     * COMPILER_FLAGS: compiler flags to be used
-#     * COVERAGE_SRC_DIR: the source directory for the coverage run
-#     * COVERAGE_EXCL: excluded files
-#
-.PHONY: coverage
-coverage: ${COVERAGE} ${ENV_DIR}/.test-requirements ${TEST_SETUP_PY}
-	rm -rf htmlcov/
-	${COVERAGE} erase
-	# For each setup.py file in TEST_SETUP_PY, run under coverage
-	$(foreach setup, ${TEST_SETUP_PY}, \
-		$(if ${TEST_PYTHON_PATH},PYTHONPATH=${TEST_PYTHON_PATH},) ${COMPILER_FLAGS} ${COVERAGE} run $(if ${COVERAGE_SRC_DIR},--source ${COVERAGE_SRC_DIR},) $(if ${COVERAGE_EXCL},--omit "${COVERAGE_EXCL}",) -a ${setup} test &&) true
-	${COVERAGE} combine
-	${COVERAGE} report -m --fail-under 100
-	${COVERAGE} html
+# To run the tests, we need to install the dependencies
```

Comment:
Not sure how this comment matches the line below, which talks about requirements.